MekHQ/src/mekhq/campaign/LogEntry.java

```diff
@@ -35,116 +35,124 @@
+	// private static final DateTimeFormatter DTF = DateTimeFormatter.ofPattern("", Locale.ENGLISH);
+
+	public LogEntry(Date date, String desc) {
+		this(date, desc, null);
+	}
+	
+	public LogEntry(Date date, String desc, String type) {
+		this.date = date;
+		this.desc = desc != null ? desc : ""; //$NON-NLS-1$
+		this.type = type;
+	}
+
+	private Date date;
+	private String desc; // non-null
+	private String type;
+
+	public Date getDate() {
+		return date;
+	}
+	
+	public void setDate(Date date) {
+		this.date = date;
+	}
+	
+	public String getDesc() {
+		return desc;
+	}
+	
+	public void setDesc(String desc) {
+		this.desc = desc != null ? desc : ""; //$NON-NLS-1$
+	}
+	
+	public String getType() {
+		return type;
+	}
+	
+	public void setType(String type) {
+		this.type = type;
+	}
+
+	@Override
+	public void writeToXml(PrintWriter pw, int indent) {
+		StringBuilder sb = new StringBuilder();
+		sb.append(MekHqXmlUtil.indentStr(indent)).append("<logEntry>"); //$NON-NLS-1$
+		if (date != null)    sb.append("<date>").append(DATE_FORMAT.format(date)).append("</date>"); //$NON-NLS-1$ //$NON-NLS-2$
+		assert desc != null; sb.append("<desc>").append(MekHqXmlUtil.escape(desc)).append("</desc>"); //$NON-NLS-1$ //$NON-NLS-2$
+		if (type != null)    sb.append("<type>").append(MekHqXmlUtil.escape(type)).append("</type>");  //$NON-NLS-1$//$NON-NLS-2$
+		sb.append("</logEntry>"); //$NON-NLS-1$
+		pw.println(sb.toString());
+	}
+
+	public static LogEntry generateInstanceFromXML(Node wn) {
+		final String METHOD_NAME = "generateInstanceFromXML(Node)"; //$NON-NLS-1$
+
+		Date   date = null;
+		String desc = null;
+		String type = null;
+
+		try {
+			NodeList nl = wn.getChildNodes();
+			for (int x = 0; x < nl.getLength(); x++) {
+				Node   node = nl.item(x);
+				String nname = node.getNodeName();
+				if (nname.equalsIgnoreCase("desc")) { //$NON-NLS-1$
+					desc = MekHqXmlUtil.unEscape(node.getTextContent());
+				} else if (nname.equalsIgnoreCase("type")) { //$NON-NLS-1$
+					type = MekHqXmlUtil.unEscape(node.getTextContent());
+				} else if (nname.equalsIgnoreCase("date")) { //$NON-NLS-1$
+					date = DATE_FORMAT.parse(node.getTextContent().trim());
+				}
+			}
+		} catch (Exception ex) {
+			MekHQ.getLogger().error(LogEntry.class, METHOD_NAME, ex);
+			return null;
+		}
+
+		return new LogEntry(date, desc, type);
+	}
+
+	@Override
+	public String toString() {
+		StringBuilder sb = new StringBuilder();
+		if (null != date) sb.append("[").append(DATE_FORMAT.format(date)).append("] "); //$NON-NLS-1$ //$NON-NLS-2$
+		sb.append(desc);
+		if (null != type) sb.append(" (").append(type).append(")");  //$NON-NLS-1$//$NON-NLS-2$
+		return sb.toString();
+	}
+
+	@Override
+	public LogEntry clone() {
+		return new LogEntry(date, desc, type);
+	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((date == null) ? 0 : date.hashCode());
```

Comment:
Consider `Objects.hashCode(object)` instead of the ternary (but only consider, do not consider this a normative statement).